action.ts

```diff
@@ -76,18 +77,34 @@ export async function runAction() {
     version = commit;
   }
   const imageWithVersion = `${imageName}:${version}`;
-  const localTag = `${orgId}/${imageWithVersion}`;
-  const imageId = await docker.build(localTag, file, additionalDockerArguments, context);
-  if (!imageId) {
-    core.setFailed('Unable build image from Dockerfile.');
-    return;
+
+  let imageId;
+  if (existingImage) {
+    imageId = existingImage;
+  } else {
+    const localTag = `${orgId}/${imageWithVersion}`;
+    imageId = await docker.build(localTag, file, additionalDockerArguments, context);
+    if (!imageId) {
+      core.setFailed('Unable build image from Dockerfile.');
+      return;
+    }
   }
 
   const remoteTag = `${registryHost}/${imageWithVersion}`;
-  const pushed = await docker.push(imageId, remoteTag);
-  if (!pushed) {
-    core.setFailed('Unable to push image to registry');
-    return;
+  if (existingImage !== remoteTag) {
```

Comment:
This feels a bit magic, but wonder what a better approach would be.

I guess the different use-cases are:
* I built an image, but now want it to be pushed to the humanitec registry (using the version from the action)
* I built and pushed an image and want to register it, but which version should we use in this case? The one provided or the one from the image 🤔

Reply:
> Currently people need to keep version: ${VERSION_TAG} and image:${VERSION_TAG} in-sync and otherwise the action fails

This sounds reasonable to me, and, if I get it properly, should also answer to the question:

> I built and pushed an image and want to register it, but which version should we use in this case? The one provided or the one from the image 🤔.